src/main/java/com/marklogic/kafka/connect/source/AbstractRowBatchBuilder.java

```diff
@@ -84,4 +91,23 @@ private void configureBatchView(Map<String, Object> parsedConfig, RowBatcher<?>
     private void onFailureHandler(RowBatchFailureListener.RowBatchFailureEvent batch, Throwable throwable) {
         logger.error("batch "+batch.getJobBatchNumber()+" failed with error: "+throwable.getMessage());
     }
+
+    protected static String injectConstraintIntoDslQuery(String originalDsl, String constrainingColumn, String constrainingValue) {
+        String constrainingPhrase = ".where(op.gt(op.col(\"" + constrainingColumn + "\"), " + constrainingValue + "))";
```

Comment:
I think we need a test for a date value, which I'm pretty sure will need to be quoted (and single quotes should work fine). And while I don't know the risks of "Optic injection attack", I think we'd want to err on the side of safety and remove any single/double quotes in both the column and the value - not that we expect them, but for safety reasons. 

A refactor to consider too for the serialized plan story (out of scope for now) - because you're unit testing this (good), and because the logic isn't trivial, I think making a "ConstraintInjector" interface with  "DslQueryConstraintInjector" and "SerializedQueryConstraintInjector" implementations would be good. That would increase readability and make unit testing easier, as you could test the logic in isolation without having to construct an RBB.

Reply:
Deferred until the serialized constraint story.

Reply:
Tests in place for the first half.